dev/src/pool.ts

```diff
@@ -207,13 +207,14 @@ export class ClientPool<T> {
 
     // Wait for all pending operations to complete before terminating.
     if (this.opCount > 0) {
-      this.operationsCompletedDeferred = new Deferred<void>();
+      this.terminateDeferred = new Deferred<void>();
```

Comment:
If you apply the previous suggestion, this line can be removed. You may want to update the comment in line 45 as well.

Reply:
removed here and added comment above to include that terminate() has to be called.